Framework/Muon/test/ApplyMuonDetectorsGroupingTest.h

```diff
@@ -0,0 +1,263 @@
+    TS_ASSERT_THROWS_NOTHING(algCompare.setProperty("Tolerance", 0.001));
+    TS_ASSERT_THROWS_NOTHING(algCompare.execute());
+    TS_ASSERT(algCompare.isExecuted());
+
+    TS_ASSERT(algCompare.getProperty("Result"));
+
+    TS_ASSERT(wsGroup);
+    if (wsGroup) {
+      TS_ASSERT_EQUALS(wsGroup->getNumberOfEntries(), 2);
+    }
+  }
+
+  void test_grouping_with_counts() {
+
+    std::string emptyString("");
+    // Populate y-values with 2.0
+    MatrixWorkspace_sptr ws =
+        WorkspaceCreationHelper::createGroupedWorkspace2D(1, 10, 0.5);
+    AnalysisDataService::Instance().addOrReplace("inputData", ws);
+
+    auto wsGroup = boost::make_shared<WorkspaceGroup>();
+    AnalysisDataService::Instance().addOrReplace("inputGroup", wsGroup);
+
+    std::string wsName = ws->getName();
+    std::string wsGroupName = (wsGroup)->getName();
+
+    TS_ASSERT_EQUALS(wsGroup->getNumberOfEntries(), 0);
+    TS_ASSERT_EQUALS(ws->getNumberHistograms(), 1);
+
+    ApplyMuonDetectorGrouping alg;
+    TS_ASSERT_THROWS_NOTHING(alg.initialize());
+    TS_ASSERT(alg.isInitialized());
+
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("InputWorkspace", ws->getName()));
+    TS_ASSERT_THROWS_NOTHING(
+        alg.setProperty("InputWorkspaceGroup", wsGroup->getName()));
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("groupName", "test"));
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("Grouping", "1,2,3,4,5"));
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("analysisType", "Counts"));
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("TimeMin", 0.0));
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("TimeMax", 30.0));
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("RebinArgs", emptyString));
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("TimeOffset", 0.0));
+    TS_ASSERT_THROWS_NOTHING(
+        alg.setProperty("SummedPeriods", std::to_string(1)));
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("SubtractedPeriods", emptyString));
+    TS_ASSERT_THROWS_NOTHING(alg.execute());
+    TS_ASSERT(alg.isExecuted());
+
+    TS_ASSERT(wsGroup->getItem("inputGroup; Group; test; Counts; #1"));
+    TS_ASSERT(wsGroup->getItem("inputGroup; Group; test; Counts; #1_Raw"));
+
+    auto wsOut = boost::dynamic_pointer_cast<MatrixWorkspace>(
+        wsGroup->getItem("inputGroup; Group; test; Counts; #1_Raw"));
+
+    TS_ASSERT_DELTA(wsOut->readX(0)[0], 0.000, 0.001);
+    TS_ASSERT_DELTA(wsOut->readX(0)[4], 2.000, 0.001);
+    TS_ASSERT_DELTA(wsOut->readX(0)[9], 4.500, 0.001);
+
+    TS_ASSERT_DELTA(wsOut->readY(0)[0], 10.000, 0.001);
+    TS_ASSERT_DELTA(wsOut->readY(0)[4], 10.000, 0.001);
+    TS_ASSERT_DELTA(wsOut->readY(0)[9], 10.000, 0.001);
+    // Quadrature errors : Sqrt(10)
+    TS_ASSERT_DELTA(wsOut->readE(0)[0], 3.162, 0.001);
+    TS_ASSERT_DELTA(wsOut->readE(0)[0], 3.162, 0.001);
+    TS_ASSERT_DELTA(wsOut->readE(0)[0], 3.162, 0.001);
+
+    TS_ASSERT(wsGroup);
+    if (wsGroup) {
+      TS_ASSERT_EQUALS(wsGroup->getNumberOfEntries(), 2);
+    }
+  }
+
```

Comment:
can we have a test that does asymmetry and one that looks at multiple period data